crates/typst/src/export/render.rs

```diff
@@ -461,10 +478,6 @@ fn render_outline_glyph(
         let top = bitmap.top;
         let bottom = top + mh;
 
-        // Premultiply the text color.
-        let Paint::Solid(color) = text.fill else { todo!() };
-        let color = bytemuck::cast(sk::ColorU8::from(color).premultiply());
-
         // Blend the glyph bitmap with the existing pixels on the canvas.
         let pixels = bytemuck::cast_slice_mut::<u8, u32>(canvas.data_mut());
         for x in left.clamp(0, cw)..right.clamp(0, cw) {
```

Comment:
This loop could be extracted into a generic function to make it more optimizable for the non-gradient path.